### PR TITLE
Jira DOC-306: Document LDAP support in new release

### PR DESCRIPTION
**Pull request by lanceleonard**
Initial check-in.  Still need to edit and add screenshots.

### CODE REVIEW

File: content/rs/security/ldap/_index.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+description: Describes how Redis Software integrates LDAP authentication and authorization.  Also describes how to enable LDAP for your deployment of Redis Software.
+weight: 25
+alwaysopen: false
+categories: ["RS"]
+aliases: /rs/administering/designing-production/security/ldap-integration/
+         /rs/security/admin-console-security/ldap/
+         /rs/security/ldap/
+---
+
+As of v6.0.20, Redis Software integrates [Lightweight Directory Access Protocol](https://en.wikipedia.org/wiki/Lightweight_Directory_Access_Protocol) (LDAP)  authentication and authorization into its role-based access controls (RBAC).  You can now use LDAP to authorize access to the admin console and to authorize database access.
```
Comment (kaitlynmichael) [inline content/rs/security/ldap/_index.md:13]: Redis Enterprise Software

File: content/rs/security/ldap/_index.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+Furthermore, you can configure LDAP roles using the admin console or the Redis Software REST API.
+
+Previously, you could enable LDAP authentication for admin console users by configuring a cluster either through the command-line or the REST API.   
+
+The cluster-based LDAP mechanism still supported in v6.0.20; however, the mechanism is deprecated and will be removed in a future update.
```
Comment (kaitlynmichael) [inline content/rs/security/ldap/_index.md]: missing "is"

File: content/rs/security/ldap/_index.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+Note that the cluster-based mechanism is not compatible with the new role-based approach. You can use either for now, but not both at the same time.
```
Comment (kaitlynmichael) [inline content/rs/security/ldap/_index.md:21]: Should this be in a note?

File: content/rs/security/ldap/_index.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+## How it works
+
+Here's how role-based LDAP integration works:
+
+1.  User signs in using their LDAP credentials.  
+
+    Based on the LDAP configuration details, the username is mapped to an LDAP Distinguished Name.
+
+1.  A simple LDAP bind is attempted using the Distinguished Name and the password.  The sign-in fails if the bind fails.
```
Comment (kaitlynmichael) [inline content/rs/security/ldap/_index.md:33]: What is an LDAP bind?

File: content/rs/security/ldap/_index.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+1.  Obtain user’s LDAP group memberships.
+
+    Using configured LDAP details, obtain a list of the user’s group memberships.
```
Comment (kaitlynmichael) [inline content/rs/security/ldap/_index.md:37]: This sentence seems a bit redundant. Do we need it?

File: content/rs/security/ldap/_index.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+For database access, the user needs to belong to an LDAP group mapped to a role listed in database’s access control list (ACL).  The rights associated with the ACL determines the suthorized level of access. 
+
+## Prerequisities 
+
+Before enabling LDAP in Redis Software, you should verify a few things:
+
+1.  You’ll need to know the LDAP groups that correspond to the levels of access you wish to authorize.  Each LDAP group will be mapped to a Redis Software access control group.
+
+1.  You’ll also need a Redis Software access control group for each LDAP group.  If you haven’t already set up role-based access controls (RBAC), you should do so before enabling LDAP.
```
Comment (kaitlynmichael) [inline content/rs/security/ldap/_index.md:53]: Add link to RBAC definition in glossary?

File: content/rs/security/ldap/cluster-based-ldap-authentication.md [hunk 7]
```diff
@@ -1,11 +1,23 @@
...
+hidden: true
+weight: 99
 alwaysopen: false
 categories: ["RS"]
-aliases: [/rs/administering/designing-production/security/ldap-integration/]
+aliases: /rs/administering/designing-production/security/ldap-integration/
+         /rs/security/admin-console-security/ldap/
 ---
+{{<warning>}}
+This article describes features that are considered deprecated.  They will be removed from a future version of Redis Software.<br/><br/>
```
Comment (kaitlynmichael) [inline content/rs/security/ldap/cluster-based-ldap-authentication.md]: Redis Enterprise Software

File: content/rs/security/ldap/cluster-based-ldap-authentication.md [hunk 8]
```diff
@@ -1,11 +1,23 @@
...
+         /rs/security/admin-console-security/ldap/
 ---
+{{<warning>}}
+This article describes features that are considered deprecated.  They will be removed from a future version of Redis Software.<br/><br/>  
+  
+As of v6.0.20, Redis Software supports role-based LDAP authentication, which can be enabled using the admin console or the Redis Software REST API.<br/><br/>  
+  
+The technique and limitations described in this article are still included with v6.0.20, however, they are considered obsolete, if favor of the new mechanism.<br/><br/>  
+  
+For now, you can continue to use cluster-based LDAP integration, however, support will be removed in a future version.  We strongly recommend migrating to role-base LDAP.
```
Comment (kaitlynmichael) [inline content/rs/security/ldap/cluster-based-ldap-authentication.md:21]: Link to new page?

File: content/rs/security/ldap/enable-role-based-ldap.md [hunk 9]
```diff
@@ -0,0 +1,87 @@
+---
+Title: Enable role-based LDAP
+description: Describes how to use Settings | LDAP to enable role-based LDAP authentication and authorization using the Redis Software admin console.
+weight: 25
+alwaysopen: false
+categories: ["RS"]
+aliases: 
+---
+
+Redis Software uses a role-based mechanism to enable LDAP authentication and authorization.
```
Comment (kaitlynmichael) [inline content/rs/security/ldap/enable-role-based-ldap.md:10]: Redis Enterprise Software

**Review by kaitlynmichael — APPROVED** [review]

File: content/rs/security/ldap/cluster-based-ldap-authentication.md (hunk 8 above)
Comment (lanceleonard) [reply content/rs/security/ldap/cluster-based-ldap-authentication.md:21]: Fixed.

File: content/rs/security/ldap/_index.md (hunk 4 above)
Comment (lanceleonard) [reply content/rs/security/ldap/_index.md]: Added link to definition.

File: content/rs/security/ldap/_index.md (hunk 2 above)
Comment (lanceleonard) [reply content/rs/security/ldap/_index.md]: Fixed

File: content/rs/security/ldap/_index.md [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+description: Describes how Redis Software integrates LDAP authentication and authorization.  Also describes how to enable LDAP for your deployment of Redis Software.
+weight: 25
+alwaysopen: false
+categories: ["RS"]
+aliases: /rs/administering/designing-production/security/ldap-integration/
+         /rs/security/admin-console-security/ldap/
+         /rs/security/ldap/
+---
+
+As of v6.0.20, Redis Enterprise Software integrates [Lightweight Directory Access Protocol](https://en.wikipedia.org/wiki/Lightweight_Directory_Access_Protocol) (LDAP) authentication and authorization into its role-based access controls (RBAC).  You can now use LDAP to authorize access to the admin console and to authorize database access.
```
Comment (AlonMagrafta) [inline content/rs/security/ldap/_index.md:13]: v6.0.20 is a new annotation, should we keep RS 6.0.20 like in other docs pages?
@lanceleonard

Comment (AlonMagrafta) [reply content/rs/security/ldap/_index.md:13]: Or we want to move towards this new annotation?

Comment (AlonMagrafta) [inline content/rs/security/ldap/_index.md:13]: Link "role-based access controls (RBAC)" to /rs/security/passwords-users-roles/ ?

File: content/rs/security/ldap/_index.md [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+
+To access the admin console, the user needs to belong to an LDAP group mapped to an Administrative role.  
+
+For database access, the user needs to belong to an LDAP group mapped to a role listed in database’s access control list (ACL).  The rights granted to the group determine the user's level of access. 
+
+## Prerequisities 
+
+Before enabling LDAP in Redis Software, you should verify a few things:
+
+1.  You’ll need to know the LDAP groups that correspond to the levels of access you wish to authorize.  Each LDAP group will be mapped to a Redis Software access control group.
```
Comment (AlonMagrafta) [inline content/rs/security/ldap/_index.md:51]: "Redis Software access control group" is kind of a new name.
I would rather use Redis Software access control role or any subset of it:
- Redis Software access control role
- Redis Software ACL
- Redis Software role

File: content/rs/security/ldap/_index.md [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+
+1.  You’ll also need a Redis Software access control group for each LDAP group.  If you haven’t already set up [role-based access controls]({{<relref "rs/security/passwords-users-roles.md##olebased-access-control">}}) (RBAC), you should do so before enabling LDAP.
```
Comment (AlonMagrafta) [inline content/rs/security/ldap/_index.md:53]: same here

File: content/rs/security/ldap/_index.md [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+
+1.  Finally, you need the the following LDAP info:
+
+    - Server URI, including host, port, and protocol details.  
+    - Certificate details for secure protocols.  
+    - Bind credentials, including Distinguished Name, password, and (optionally) client public and private keys for certificate authentication.  
+    - Authentication query details, whether template or query.  
+    - Authorization query details, whether template or query.
```
Comment (AlonMagrafta) [inline content/rs/security/ldap/_index.md:61]:
```suggestion
    - Authorization query details, whether attribute or query.  
```

Comment (lanceleonard) [reply content/rs/security/ldap/_index.md]: Fixed.

File: content/rs/security/ldap/_index.md (hunk 12 above)
Comment (lanceleonard) [reply content/rs/security/ldap/_index.md]: Changed group to role; prefer to leave "map" as a verb overall for this feature.

File: content/rs/security/ldap/_index.md (hunk 11 above)
Comment (lanceleonard) [reply content/rs/security/ldap/_index.md]: Changed to "role" to match UI.

File: content/rs/security/ldap/_index.md [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+         /rs/security/ldap/
+---
+
+As of version 6.0.20, Redis Enterprise Software integrates [Lightweight Directory Access Protocol](https://en.wikipedia.org/wiki/Lightweight_Directory_Access_Protocol) (LDAP) authentication and authorization into its [role-based access controls]({{<relref "rs/security/passwords-users-roles.md##olebased-access-control">}}) (RBAC).  You can now use LDAP to authorize access to the admin console and to authorize database access.
+
+Furthermore, you can configure LDAP roles using the admin console or the Redis Software REST API.
+
+Previously, you could enable LDAP authentication for admin console users by [configuring a cluster]({{< relref "/rs/security/ldap/cluster-based-ldap-authentication.md" >}}) either through the command-line or the REST API.   
+
+The cluster-based LDAP mechanism is supported in v6.0.20; however, the mechanism is deprecated and will be removed in a future update.
```
Comment (AlonMagrafta) [inline content/rs/security/ldap/_index.md:19]: v6.0.20 / RS 6.0.20? Up to you guys :)

File: content/rs/security/ldap/cluster-based-ldap-authentication.md [hunk 15]
```diff
@@ -1,17 +1,32 @@
...
+aliases: [
+    "/rs/administering/designing-production/security/ldap-integration/",
+    "/rs/security/admin-console-security/ldap/",
+    "/rs/security/admin-console-security/ldap.md"
+]
 ---
+{{<warning>}}
+This article describes features that are considered deprecated.  They will be removed in a future release of Redis Enterprise Software.<br/><br/>  
+  
+As of v6.0.20, Redis Software supports role-based [LDAP authentication]({{< relref "/rs/security/ldap/" >}}), which can be enabled using the admin console or the Redis Software REST API.<br/><br/>
```
Comment (AlonMagrafta) [inline content/rs/security/ldap/cluster-based-ldap-authentication.md:17]: v6.0.20 / RS 6.0.20? Up to you guys :)

File: content/rs/security/ldap/cluster-based-ldap-authentication.md [hunk 16]
```diff
@@ -1,17 +1,32 @@
...
+    "/rs/security/admin-console-security/ldap/",
+    "/rs/security/admin-console-security/ldap.md"
+]
 ---
+{{<warning>}}
+This article describes features that are considered deprecated.  They will be removed in a future release of Redis Enterprise Software.<br/><br/>  
+  
+As of v6.0.20, Redis Software supports role-based [LDAP authentication]({{< relref "/rs/security/ldap/" >}}), which can be enabled using the admin console or the Redis Software REST API.<br/><br/>  
+  
+The features described in this article are still included with v6.0.20; however, they are considered obsolete and have been replaced by [role-based LDAP authentication]({{< relref "/rs/security/ldap/" >}}).<br/><br/>
```
Comment (AlonMagrafta) [inline content/rs/security/ldap/cluster-based-ldap-authentication.md:19]: v6.0.20 / RS 6.0.20? Up to you guys :)

File: content/rs/security/ldap/migrate-to-role-based-ldap.md [hunk 17]
```diff
@@ -0,0 +1,63 @@
+---
+Title: Migrate to role-based LDAP
+description: Describes how to migrate existing cluster-based LDAP deployments to role-based LDAP.
+weight: 55
+alwaysopen: false
+categories: ["RS"]
+aliases: 
+---
+
+As of v6.0.20, Redis Enterprise Software supports two LDAP authentication mechanisms: the [cluster-based mechanism]({{< relref "/rs/security/ldap/cluster-based-ldap-authentication.md" >}}) supported in earlier versions and a [role-based mechanism]({{< relref "/rs/security/ldap/" >}}).
```
Comment (AlonMagrafta) [inline content/rs/security/ldap/migrate-to-role-based-ldap.md:10]: v6.0.20 / RS 6.0.20? Up to you guys :)

**Review by AlonMagrafta — APPROVED** [review]
small notes on the v6.0.20. Up to you.
Perfect to go!

**Pull request merged**